src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
The client needs one (and exactly one) endpoint to work with.
Can we make the endpoint part of the client? And not a pointer that needs to be malloced.

Also, test all malloc for failure (NULL returned).